src/app/layout/qgslayoutdesignerdialog.cpp

```diff
@@ -4146,6 +4152,11 @@ void QgsLayoutDesignerDialog::activateNewItemCreationTool( int id, bool nodeBase
     if ( mView )
       mView->setTool( mAddNodeItemTool );
   }
+
+  if ( mLayout )
+  {
+    mLayout->deselectAll();
+  }
```

Comment:
The rationale was that pressing Del while drawing a new node item currently remove ve the previously selected item (since at this point it is still selected). But I agree, deselection should be deferred until the actual drawing starts.